tests/executables_src/testModules.cc

```diff
@@ -1081,4 +1082,128 @@ namespace Tests::testModules {
+    ctk::ScalarOutput<double> someOutput{this, "someOutput", "", "", {}};
+
+    SlashedGroup someGroup{this, "someGroup/", ""};
+
+    struct AnotherGroup : ctk::VariableGroup {
+      using ctk::VariableGroup::VariableGroup;
+      ctk::ScalarPushInput<uint8_t> foo{this, "foo", "", "", {}};
+
+      // variable names ending in slashes will produce a ctk::logic_error
+      ctk::ScalarPushInput<uint8_t> slash{this, "slash", "", "", {}};
+      ctk::ArrayPushInput<int64_t> slashArray{this, "array", "", 16, "", {}};
+
+    } anotherGroup{this, "anotherGroupName////", ""};
+
+    void mainLoop() override {}
+  };
+
+  // module group
+
+  struct SlashApp : public ctk::Application {
+    SlashApp() : Application("slashApp") {}
+    ~SlashApp() override { shutdown(); }
+
+    SlashModule slashModule{this, "slashModule/", ""};
+    struct slashModGroup : ctk::ModuleGroup {
+      using ctk::ModuleGroup::ModuleGroup;
+      SlashModule nestedSlashModule{this, "nestedSlashModule//", ""};
+    } slashModGroup{this, "slashModGroupName/", ""};
+  };
+
+  BOOST_AUTO_TEST_CASE(test_trailingSlashes) {
+    std::cout << "***************************************************************"
+                 "******************************************************"
+              << std::endl;
+    std::cout << "==> test_trailingSlashes" << std::endl;
+    std::cout << std::endl;
+
+    SlashApp app;
+    ctk::TestFacility tf{app};
+    tf.runApplication();
+
+    // check module name
+    BOOST_CHECK(app.slashModule.getName() == "slashModule");
+    // check variable group names
+    BOOST_CHECK(app.slashModule.someGroup.getName() == "someGroup");
+    // anotherGroupName has many slashes
+    BOOST_CHECK(app.slashModule.anotherGroup.getName() == "anotherGroupName");
+
+    // check module group names
+    BOOST_CHECK(app.slashModGroup.getName() == "slashModGroupName");
+    BOOST_CHECK(app.slashModGroup.nestedSlashModule.getName() == "nestedSlashModule");
+
+    // check variables
+    // variables will return a fully qualified name:
+    BOOST_CHECK(app.slashModule.someInput.getName() == "/slashModule/nameOfSomeInput");
+    BOOST_CHECK(app.slashModule.someOutput.getName() == "/slashModule/someOutput");
+    BOOST_CHECK(app.slashModule.someGroup.inGroup.getName() == "/slashModule/someGroup/inGroup");
+    BOOST_CHECK(app.slashModule.someGroup.alsoInGroup.getName() == "/slashModule/someGroup/alsoInGroup");
+    BOOST_CHECK(app.slashModule.anotherGroup.foo.getName() == "/slashModule/anotherGroupName/foo");
+    BOOST_CHECK(app.slashModule.anotherGroup.slash.getName() == "/slashModule/anotherGroupName/slash");
+    BOOST_CHECK(app.slashModule.anotherGroup.slashArray.getName() == "/slashModule/anotherGroupName/array");
+  }
+
+  /*********************************************************************************************************************/ /* test tailing slashes in scallar variable name */
```

Comment:
typo: "scallar variable" -> "scalar variable"

also I think the comment should go into a separate line?